src/Symfony/Component/Serializer/Normalizer/ArrayDenormalizer.php

```diff
@@ -36,13 +37,14 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
         if (null === $this->denormalizer) {
             throw new BadMethodCallException('Please set a denormalizer before calling denormalize()!');
         }
-        if (!\is_array($data)) {
-            throw new InvalidArgumentException('Data expected to be an array, '.get_debug_type($data).' given.');
-        }
         if (!str_ends_with($type, '[]')) {
             throw new InvalidArgumentException('Unsupported class: '.$type);
         }
 
+        if (!\is_array($data)) {
```

Comment:
Why inverting the order with the check on the type? Calling `is_array()` is cheaper than calling `str_ends_with()`, so it should be done first (especially because this method is ok the hot path and can be called hundreds of times).

Reply:
@NorthBlue333 Can you answer this question please?

Reply:
Indeed, IHMO, there is no need to change the order of checks there.

Reply:
Hi, sorry did not have much time for those PRs. I will take a look during the end of the week.